### ISSUE TITLE
sheets: How to get real time updates for spreadsheets?

### ISSUE THREAD

**Issue by eminano** [googleapis/google-api-go-client#444]
Hi! 

We're trying to refresh google sheets in real time and it's proving difficult. 

The [push notifications ](https://developers.google.com/drive/api/v3/push)(webhooks) of the google drive API are only sent as frequently as every 3 min (they are batched after the first update), which isn't frequent enough for real time updates. 

Retrieving the ```modifiedTime``` for a spreadsheet using the google drive API seems to hit the same data in the background, which is only refreshed every ~3min.

Using the google sheets API is not an option since the modified time information is no longer supported in the latest version (v4). 

What API are we supposed to use if we want real time update information for spreadsheets?

Thanks in advance,
Esther

**Comment by tbpg** [comment]
cc @asrivas (can't assign for some reason).

**Comment by sqrrrl** [comment]
TL;DR - There's no API for real time updates at the moment.

Best you can do is one of 3 options:

* As you're already aware, Drive an push updates for sheets but changes are coalesced and only sent around every 3 minutes or so.
* Poill, as you also mentioned. Even without the modified time in the sheet, you can still detect changes in the sheet by comparing the raw response. Main downside is it chews up quota faster as you shorten the polling intervals to get closer to real time.
* Attach a script/add-on to the sheet and use [onEdit trigger](https://developers.google.com/apps-script/guides/triggers). You can either do the processing in apps script or use UrlFetchApp to call a webhook or push to pub/sub to process the update elsewhere. Limitations are that onEdit is only called for interactive sessions (user making edits in the UI,) not for programmatic updates via the API or apps script. Also can be a little lossy if user editing multiple cells quickly. You'll still get an update, but the range details in the event might not capture everything that changed.

If you can share more details about the use case it would be helpful in prioritizing this as a feature request.

**Comment by codyoss** [comment]
Closing this for now due to inactivity. Please feel free to reopen if you have some more details.
